zebra-state/src/service/non_finalized_state.rs

```diff
@@ -480,17 +480,7 @@ impl NonFinalizedState {
                 let fork_chain = self
                     .chain_set
                     .iter()
-                    .find_map(|chain| {
-                        chain
-                            .fork(
-                                parent_hash,
-                                sprout_note_commitment_tree.clone(),
-                                sapling_note_commitment_tree.clone(),
-                                orchard_note_commitment_tree.clone(),
-                                history_tree.clone(),
-                            )
-                            .transpose()
-                    })
+                    .find_map(|chain| chain.fork(parent_hash).transpose())
                     .transpose()?
```

Comment:
We can get rid of these two transpose calls if `fork` returns `Option<Self>`:

```suggestion
                    .find_map(|chain| chain.fork(parent_hash))
```

Reply:
Thanks, fixed in 7ea4bf8db9